### PR TITLE
St2 deploy script with RHEL6 support using IUS repository for python27

### PR DESCRIPTION
**Pull request by emptywee**
The first quick&dirty st2_deploy.sh fix to make it work smoothly on RHEL6-family linux distros. Tested on RHEL6.4+ and CentOS 7. Worked like a charm for me. Feel free to review and refactor if you feel the need.

### CODE REVIEW

**Comment by DoriftoShoes** [comment]
This looks good.  I'll run it tonight to validate everything is GTG then merge as long as I don't find any issues.  Thanks!

**Comment by armab** [comment]
Worked for me under CentOS 7

File: tools/st2_deploy.sh [hunk 1]
```diff
@@ -381,6 +451,134 @@ mistral_systemd
...
+{
+echo "Setting up initd script for Mistral..."
+initd=/etc/init.d/mistral
+if [ -e "$initd" ]; then
+    rm $initd
+fi
+touch $initd
+chmod +x $initd
+
+cat <<mistral_initd >$initd
```
Comment (Kami) [inline tools/st2_deploy.sh]: Really wish we could store this file somewhere outside the deploy script... :)

Comment (DoriftoShoes) [inline tools/st2_deploy.sh]: We have it in Puppet.  Packages will come some day :)

File: tools/st2_deploy.sh [hunk 2]
```diff
@@ -123,10 +135,37 @@ if [[ "$DEBTEST" == "Ubuntu" ]]; then
...
+    chkconfig firewalld off
+    PIP="pip2.7"
+    VIRTUALENV="virtualenv-2.7"
+    PYTHON="python2.7"
+    YUM_PYTHON=$(join " " ${YUM_PYTHON_6[@]})
+  else
+    echo "Unknown RHEL-family... Aborting install."
+    exit 2
+  fi
+
```
Comment (DoriftoShoes) [inline tools/st2_deploy.sh]: This will break Fedora deployments since they won't match on the 6/7.  I'll look at refining this a bit.

**Comment by DoriftoShoes** [comment]
Hmmm... `requiretty` didn't get disabled when I tested this on RHEL6 tonight.

**Comment by DoriftoShoes** [comment]
Overall great PR.  Thanks!  I validated on RHEL6 and RHEL7, the only missing piece is disabling requiretty in the sudoers file and this PR to fix the python binary is a blocker:

https://github.com/StackStorm/st2/pull/1748

Once that PR gets merged and the conditional logic for Fedora is added I'll merge this puppy in.

File: tools/st2_deploy.sh (hunk 2 above)
Comment (emptywee) [inline tools/st2_deploy.sh]: Ooops, forgot about Fedora. This is easy to fix though. I am not really familiar with Fedora's versions and their python versions usage.

**Comment by emptywee** [comment]
Sure, you are welcome. I am happy to help/contribute.

**Pull request merged**